src/semafour.cc

```diff
@@ -257,6 +325,50 @@ void Semafour::Close(const v8::FunctionCallbackInfo<v8::Value>& args) {
 }
 
 
+void Semafour::TryWait(const v8::FunctionCallbackInfo<v8::Value>& args) {
+  Semafour* sem = UNWRAP(Semafour);
+  Isolate* isolate = args.GetIsolate();
+  int r;
+
+  if (args.Length() == 0) {
+    // Handle synchronous case.
+    r = sem->TryWait();
+
+    if (r != TRYWAIT_LOCKED && r != TRYWAIT_OK) {
+      THROW_UV(r, "sem_trywait");
```

Comment:
The `THROW` macros were updated to take three arguments.

Reply:
Rah, I'm such an idiot sometimes!